### ISSUE TITLE
can't build the app

### ISSUE THREAD

**Issue by devin-lo** [DoaJCBlogger/Talk32#12]
Hi, I'm trying to build the app in Windows 10 Pro so that I can use Talk32 on my WinXP machine.
But it seems that I can't even build it?

![image](https://user-images.githubusercontent.com/75116968/226219725-10555946-c9a9-4dfc-8261-71f439c860d9.png)

I have installed Windows SDK 7.1 by first uninstalling VC++ 2010 Redistributables, then installing the SDK, then installing the patch, then finally reinstalling the redistributables.

Is there something I did wrong? Or can you just send me one that's already built please?

**Comment by DoaJCBlogger** [comment]
You're missing ammintrin.h. This is a known issue with the VC++ 2010 compiler. I've attached a copy. You should also install the SP1 update.
[ammintrin.zip](https://github.com/DoaJCBlogger/Talk32/files/11012686/ammintrin.zip)
[Fix problem when mex .pdf](https://github.com/DoaJCBlogger/Talk32/files/11012692/Fix.problem.when.mex.pdf)
[fatal error C1083_ Cannot open include file_ ‘ammintrin.pdf](https://github.com/DoaJCBlogger/Talk32/files/11012696/fatal.error.C1083_.Cannot.open.include.file_.ammintrin.pdf)

**Comment by devin-lo** [comment]
Hey! That helped to build it successfully, thank you for the quick reply!
I had seen that issue on StackOverflow, but the way they linked to the upload on the Mathworks forum was broken.

Now, I'm having this issue in WinXP:
![Imgur](https://imgur.com/5k2USY8)

**Comment by DoaJCBlogger** [comment]
I use the Windows SDK 7.1, not Visual Studio. If you want Visual Studio, I trust files.dog. You should check their MSDN collection. I don't know exactly what's causing the error but you might try deleting talk32.exe and building it again. Sometimes the manifest doesn't get added (the final step with "mt").

**Comment by devin-lo** [comment]
Rebuilding it didn't fix it :( I tried doing the steps from the bat individually.

It opens fine in Windows 10 but not on XP.

Actually, is there any way to build directly on Windows XP? Does Windows SDK 7.1 even work on XP itself?

**Comment by DoaJCBlogger** [comment]
Yes, it runs on XP and that's partly why I use it. Would you like me to just give you a copy of the latest version?

**Comment by DoaJCBlogger** [comment]
If it still doesn't work after building it on Windows XP, send me a screenshot of all the files in the folder that talk32.exe is in so I can make sure that you're not missing anything.

**Comment by devin-lo** [comment]
Tried to build in XP, yeah still not working.

![Imgur](https://i.imgur.com/z7Wbtqp.png)

If you can't figure it out, yes please send a copy of the latest version for me to try opening.

**Comment by DoaJCBlogger** [comment]
This copy is able to start on Windows 8.1. Let me know if it works for you. Also, the messaging parts aren't ready yet but if you want to use the logger, you're going to have to create a JSON config file. I can help you do that tomorrow if you want.

[talk32_20230320.zip](https://github.com/DoaJCBlogger/Talk32/files/11024430/talk32_20230320.zip)

**Comment by devin-lo** [comment]
It works!! Thank you so much :)

**Comment by devin-lo** [comment]
Oh, now I have this error that there's no cache folder:
![Imgur1](https://i.imgur.com/RxnneqP.png)

Providing the console log for reference:
![Imgur2](https://i.imgur.com/Yrivo6K.png)

**Comment by DoaJCBlogger** [comment]
Extract this to a folder called "Talk32" in your local AppData folder. Here is where it will be. There is also a CHM help file in the "docs" folder that explains the contents if you want to set up the logger, which is the only part that's ready for use.

Windows XP: C:\Documents and Settings\[username]\Local Settings\Application Data\
Windows 7 and higher: C:\Users\[username]\AppData\Local\

Also, be careful about uploading screenshots of the log since it might contain your authentication token which gives people full access to your account even if you have 2-factor authentication.

[talk32_cache.zip](https://github.com/DoaJCBlogger/Talk32/files/11057635/talk32_cache.zip)

**Comment by devin-lo** [comment]
Hooray, adding the cache folder worked! I managed to get past the login screen now.

However, logging doesn't seem to work, as any positive number of entries I put into the logging_servers array in config.json makes Talk32 crash upon opening :(

Also, noted, I deleted the screenshot.

Thanks for putting up with me and my unintentional QA testing!

**Comment by DoaJCBlogger** [comment]
Can you send your config.json file?

**Comment by devin-lo** [comment]
[config.txt](https://github.com/DoaJCBlogger/Talk32/files/11075688/config.txt)
Uploaded as TXT instead of JSON.
Edits were made to this file exclusively in XP's Notepad.

**Comment by DoaJCBlogger** [comment]
Can you send me an invitation to "omg yesss fortnie" so I can see if it's something about the server?

**Comment by DoaJCBlogger** [comment]
It took me a while to realize that it might be awkward to explain why I joined that server so we don't have to do that if you don't want to. I'm able to use the logger nonstop for multiple consecutive days without any issues so I don't know what's wrong.

**Comment by DoaJCBlogger** [comment]
One more thing: try adding an "assets" element to the server objects, like this:

`"assets": "E:\\Documents\\discord\\assets\\monero\\"`

**Comment by DoaJCBlogger** [comment]
This issue is years-old and should not happen in the current version so I'm going to close it. You can create a new one if it still happens
